_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -725,6 +725,73 @@ namespace
         return mfxU32(384 * GetMaxMbps(par) / frameRate / GetMinCr(par.mfx.CodecLevel));
     }
 
+    mfxStatus CheckMaxFrameSize(MfxVideoParam & par, ENCODE_CAPS const & hwCaps)
+    {
+        mfxExtCodingOption2 * extOpt2 = GetExtBuffer(par);
+        mfxExtCodingOption3 * extOpt3 = GetExtBuffer(par);
+
+        if (extOpt2->MaxFrameSize == 0 && extOpt3->MaxFrameSizeI == 0 && extOpt3->MaxFrameSizeP == 0)
+            return MFX_ERR_NONE;
+
+        bool changed = false;
+        bool unsupported = false;
+
+        bool IsEnabledSwBrc = bRateControlLA(par.mfx.RateControlMethod) || (IsOn(extOpt2->ExtBRC) && (par.mfx.RateControlMethod == MFX_RATECONTROL_CBR || par.mfx.RateControlMethod == MFX_RATECONTROL_VBR));
+
+        if ((par.mfx.RateControlMethod == MFX_RATECONTROL_CBR && // max frame size supported only for VBR based methods
+            par.mfx.RateControlMethod == MFX_RATECONTROL_CQP) ||
```

Comment:
Looks like a bug. (RCM==2 && RCM==3) is always false.

Reply:
@onabiull : is this a bug which regressed master? @lakulako : we separated out these cherry-picks specifically because there was some issues identified with it.... I hope you've spotted one...

Reply:
@dvrogozh , @onabiull , yes it's a bug. I have submitted PR for fix it in master: https://github.com/Intel-Media-SDK/MediaSDK/pull/1180